sklearn/linear_model/coordinate_descent.py

```diff
@@ -559,31 +560,41 @@ class ElasticNet(LinearModel, RegressorMixin):
         :class:`sklearn.preprocessing.StandardScaler` before calling ``fit``
         on an estimator with ``normalize=False``.
 
-    precompute : True | False | array-like
-        Whether to use a precomputed Gram matrix to speed up
-        calculations. The Gram matrix can also be passed as argument.
-        For sparse input this option is always ``True`` to preserve sparsity.
+    copy_X : boolean, optional, default True
+        If ``True``, X will be copied; else, it may be overwritten.
 
     max_iter : int, optional
         The maximum number of iterations
 
-    copy_X : boolean, optional, default True
-        If ``True``, X will be copied; else, it may be overwritten.
-
     tol : float, optional
         The tolerance for the optimization: if the updates are
         smaller than ``tol``, the optimization code checks the
         dual gap for optimality and continues until it is smaller
         than ``tol``.
 
+    solver : str, {'auto', 'cd', 'saga'}, \
+             optional (default='cd').
+
+        Algorithm to use in the optimization problem.
+
+        - 'auto' chooses the solver automatically based on the type of data.
+          If the data is F-contiguous or a sparse 'csc' matrix, it chooses
```

Comment:
This seems quite an obscure criterion... Is the memory order of the data really sensible criteria to say one solver is more appropriate than the other?

Reply:
If you have a large dataset X, reordering from C to F or vice versa with a full copy of the array X might be a heavy (memory) and unwanted operation. The nice point is that 'cd' needs F- and 'saga' needs C-ordered arrays. Furthermore, without benchmarks, we can't tell which solver is better suited for a given X, so I came up with this criterion. I can easily change 'auto' to 'cd' or remove 'auto'. It is just a suggestion.

Reply:
I'd be inclined to remove auto, but would be interested to hear from @agramfort

Reply:
alternatively, those benchmarks we don't have yet, would be an argument for `auto`. Would you be very nice and run some benchmarks on this @lorentzenchr ? It may turn out to be the case that the copy is not that significant in most cases compared to the rest of the method afterall.

Reply:
+1 for removing auto for now but I would be very interested in some extensive benchmarks to guide future decisions on "smart" solver selection.

Reply:
Benchmarks based on bench_glmnet.py indicate that coordinate descent is the clear winner, regardless of the contiguity of the data. So, ``auto`` will be removed and ``cd`` the default solver.